chain/src/consume_unverified.rs

```diff
@@ -1,414 +1,261 @@
+                    },
+                    Err(err) => {
+                        info!("truncate_block_tx has been closed,err: {}", err);
+                        return;
+                    },
+                },
+                recv(self.stop_rx) -> _ => {
+                    info!("consume_unverified_blocks thread received exit signal, exit now");
+                    break;
                 }
-            })
-            .expect("Start ChainService failed");
-
-        register_thread("ChainService", chain_jh);
 
-        ChainController::new(process_block_sender, truncate_sender)
-    }
-
-    fn make_fork_for_truncate(&self, target: &HeaderView, current_tip: &HeaderView) -> ForkChanges {
-        let mut fork = ForkChanges::default();
-        let store = self.shared.store();
-        for bn in (target.number() + 1)..=current_tip.number() {
-            let hash = store.get_block_hash(bn).expect("index checked");
-            let old_block = store.get_block(&hash).expect("index checked");
-            fork.detached_blocks.push_back(old_block);
+            }
         }
-        is_sorted_assert(&fork);
-        fork
     }
+}
 
-    // Truncate the main chain
-    // Use for testing only, can only truncate less than 50000 blocks each time
-    pub(crate) fn truncate(&mut self, target_tip_hash: &Byte32) -> Result<(), Error> {
-        let snapshot = Arc::clone(&self.shared.snapshot());
-        assert!(snapshot.is_main_chain(target_tip_hash));
-
-        let target_tip_header = snapshot.get_block_header(target_tip_hash).expect("checked");
-        let target_block_ext = snapshot.get_block_ext(target_tip_hash).expect("checked");
-        let target_epoch_ext = snapshot
-            .get_block_epoch_index(target_tip_hash)
-            .and_then(|index| snapshot.get_epoch_ext(&index))
-            .expect("checked");
-        let origin_proposals = snapshot.proposals();
-
-        let block_count = snapshot
-            .tip_header()
-            .number()
-            .saturating_sub(target_tip_header.number());
-
-        if block_count > 5_0000 {
-            let err = format!(
-                "trying to truncate too many blocks: {}, exceed 50000",
-                block_count
-            );
-            return Err(InternalErrorKind::Database.other(err).into());
-        }
-        let mut fork = self.make_fork_for_truncate(&target_tip_header, snapshot.tip_header());
+impl ConsumeUnverifiedBlockProcessor {
+    pub(crate) fn consume_unverified_blocks(&mut self, unverified_block: UnverifiedBlock) {
+        let UnverifiedBlock {
+            block,
+            switch,
+            verify_callback,
+            parent_header,
+        } = unverified_block;
+        let block_hash = block.hash();
+        // process this unverified block
+        let verify_result = self.verify_block(&block, &parent_header, switch);
+        match &verify_result {
+            Ok(_) => {
+                let log_now = std::time::Instant::now();
```

Comment:
can we add a assertion here? since `verfiy_block` may return `OK(true)` or `OK(false)`, and `true` means it's a verified block, `false` means it's a unverified block, from the name of this function, seems it's expecting `false`?

Reply:
Both `Ok(true)` and `Ok(false)` means it's a valid block.
```rust
/// VerifyResult is the result type to represent the result of block verification
///
/// Ok(true) : it's a newly verified block
/// Ok(false): it's a block which has been verified before
/// Err(err) : it's a block which failed to verify
pub type VerifyResult = Result<bool, Error>;
```